webapp/packages/plugin-connections/src/ConnectionForm/Options/ConnectionOptionsTabService.ts

```diff
@@ -331,12 +337,8 @@ export class ConnectionOptionsTabService extends Bootstrap {
       );
     }
 
-    if (driver.mainProperties.length > 0) {
-      tempConfig.mainProperties = this.prepareDynamicProperties(
-        driver.mainProperties,
-        toJS(state.config.mainProperties),
-        tempConfig.configurationType,
-      );
+    if (tempConfig.configurationType === DriverConfigurationType.Custom && !!driver.mainProperties?.length) {
+      tempConfig.mainProperties = this.prepareDynamicProperties(driver.mainProperties, tempConfig.mainProperties, tempConfig.configurationType);
     }
```

Comment:
i will suggest to use `switch(tempConfig.configurationType)` to exclude unwanted effects from different types of configuration so you can be sure that use send only state related to selected configuration

for example:
`tempConfig.mainProperties = toJS(state.config.mainProperties);` always set

it can reflect on:
```
if (tempConfig.configurationType === DriverConfigurationType.Custom && !!driver.mainProperties?.length) {
      tempConfig.mainProperties = this.prepareDynamicProperties(driver.mainProperties, tempConfig.mainProperties, tempConfig.configurationType);
    }
```